### PR TITLE
Added feature that sort the results of `logical_select` in descending…

### PR DESCRIPTION
**Pull request by komainu8**
… order.

### CODE REVIEW

File: lib/mrb/scripts/table.rb [hunk 1]
```diff
@@ -102,7 +102,7 @@ def ensure_sort_key(key, converted_keys)
         order    = key[:order] || order
         offset   = key[:offset] || offset
       else
-        key_name = key
+        key_name = key.dup
```
Comment (kou) [inline lib/mrb/scripts/table.rb]: We should not use `#dup`. We should stop to update `key_name` directory by `key_name[0] = ""` instead. `key_name = key_name[1..-1]` is alternative way.

File: plugins/sharding/logical_select.rb [hunk 2]
```diff
@@ -152,6 +152,11 @@ def write_records(writer, context)
             :offset => current_offset,
             :limit => current_limit,
           }
+          context.sort_keys.each do |sort_key|
+            if sort_key.match("-timestamp")
```
Comment (kou) [inline plugins/sharding/logical_select.rb]: We shouldn't assume `timestamp` as sort key.
It should be `sort_key.start_with?("-")`.

Comment (kou) [reply plugins/sharding/logical_select.rb]: If `sort_keys` have two `-XXX`, `result_sets` isn't reversed.
It should be:

```ruby
if context.sort_keys.any? {|sort_key| sort_key.start_with?("-")}
  result_sets = result_sets.reverse
end
```

or something.

**Comment by komainu8** [comment]
@kou Thank you for your comments.
I've fied them below commit.
https://github.com/groonga/groonga/pull/773/commits/abcb4f41a6ae47c30b110c62710204bf9f2446ab

**Comment by kou** [comment]
Merged.

**Pull request merged**